source/application/DbgHelp/DbgHelp.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2016- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#pragma once
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "DbgHelpStackWalkCallback.h"
+
+class DbgHelpPrivate;
+
+class DbgHelp
+{
+public:
+    DbgHelp();
+    ~DbgHelp();
+
+    void AddProcess(HANDLE process, DWORD process_id);

```

Comment:
Slow reply, but finally looked into this. You want to create DbgHelp only once, and before any debugged processes have started. This means that you need (a) to handle more than one handle, and (2), the handles aren't available at the time of DbgHelp creation.